src/matrixrtc/MembershipManager.ts

```diff
@@ -997,14 +1017,64 @@ export class MembershipManager
     }
 }
 
-function createInsertActionUpdate(type: MembershipActionType, offset?: number): ActionUpdate {
-    return {
-        insert: [{ ts: Date.now() + (offset ?? 0), type }],
-    };
-}
+/**
+ * Implementation of the Membership manager that uses sticky events
+ * rather than state events.
+ */
+export class StickyEventMembershipManager extends MembershipManager {
+    public constructor(
+        joinConfig: (SessionConfig & MembershipConfig) | undefined,
+        room: Pick<Room, "getLiveTimeline" | "roomId" | "getVersion">,
+        private readonly clientWithSticky: Pick<
+            MatrixClient,
+            | "getUserId"
+            | "getDeviceId"
+            | "sendStateEvent"
+            | "_unstable_sendDelayedStateEvent"
+            | "_unstable_updateDelayedEvent"
+            | "_unstable_sendStickyEvent"
+            | "_unstable_sendStickyDelayedEvent"
+        >,
+        getOldestMembership: () => CallMembership | undefined,
```

Comment:
maybe worth defining a type `MembershipManagerClient`
with 
```
| "getUserId"
| "getDeviceId"
| "sendStateEvent"
| "_unstable_sendDelayedStateEvent"
| "_unstable_updateDelayedEvent"
```

and then doing:

```suggestion
        private readonly clientWithSticky: MembershipManagerClient & Pick<
            | "_unstable_sendStickyEvent"
            | "_unstable_sendStickyDelayedEvent"
        >,
        getOldestMembership: () => CallMembership | undefined,
```